### DISCUSSION TITLE
questions on datasource and colors

### DISCUSSION THREAD

**Discussion by Diegomarino80** [team-moeller/better-access-charts#154] [Q&A]
Hi, i just discoverd this beauty and i'm amazed.
I started to learn 3 years ago access, now i can code vba pretty good but I'm self-taught, so I probably suck a bit however.

1) I read that you can make an image to use in reports, what's the process? 
!!EDIT i think i found the sub: Private Sub SaveChartAsImage(ByRef ImageFileName As String)

2) i can't find (as i saw in viedos) how to change colors of values, and, expecially i'd like to pick two colors in my pie chart (red and green) with only two values, as you can see in the pic.

I think for colors i can use "myChart.Chart.ColorSchema.Name = "brewer.Paired12" but is it possible to pick colors individually?

![image](https://github.com/team-moeller/better-access-charts/assets/146032892/97b5f69a-3299-4691-b8ea-c5ecccd14651)

3) i changed the datasource to a query i made, but i got an error "element not found"
i made this
`    With myChart.DataSource
         .ObjectName = "qryAnalisioperazionegrafico"
        .LabelFieldName = "DataLabel"
        .DataFieldNames = Array("Costi", "Redditività")
    End With`

is it correct? thanks
i hope you can answer me, thanks

**Comment by team-moeller** [comment]
Hi @Diegomarino80 ,
1. question: you found the answer by yourself
2. question: You will have to program the individual color selection yourself. BAC doesn't support this in the way you want.
3. question: The code fragment itself looks right. I think the best way is to set a breakpoint and step line by line through the code to find the place, where the error happens.
Good luck!

**Comment by Diegomarino80** [comment]
thanks very much.
Just to take the right way, about individual colors, where is the part is should change? 

EDIT well, don't mind, some colorschema is good.
I'm not able to set a colorschema however.

`Private Sub Comando1_Click()
   Dim MyChart As BAC_Chart
   
   Set MyChart = BAC.Chart(Me.EdgeBrowser6)
   
   MyChart.ChartType = Pie
   
   With MyChart.DataSource
      .ObjectName = "qryAnalisiOperazioneGrafico"
      .LabelFieldName = "costiricavi"
      .DataFieldNames = "Valore"
   End With

   'titolo
   MyChart.Title.Show = True
   MyChart.Title.Position = posTop
   MyChart.Title.Text = "Costi/Ricavi"
   MyChart.Title.Font.Size = 25
   '---------
   MyChart.DisplayIn = chSystemBrowser
   MyChart.CreateChart
   MyChart.Chart.ColorSchema.Name = "Brewer.Paired12"

End Sub

------
  this part gives me an error: MyChart.Chart.ColorSchema.Name = "Brewer.Paired12", "Impossible to find the method". 
  I'm not able to do that.
EDIT, after a long search i found that autocolor plugin is now instead of colorscheme, but i can't find how to choose a schema 
-------

**and the last question: sometimes the chart do not appears in my custom form, i have to open the Frm_demo and then it appears. Maybe cause it reset something?**
this is my copy of BAC with some changes, you can open my form from the startup

[Better-Access-Charts.zip](https://github.com/team-moeller/better-access-charts/files/12763392/Better-Access-Charts.zip)
Thanks

**Reply by team-moeller** [reply]
Hi @Diegomarino80 ,
thank you for the sample database. I did some investigations, and I found a small bug in the code of BAC.
See: [Failed to load chart.js library (404 Not Found)](https://github.com/team-moeller/better-access-charts/issues/156)
I work on a solution right now.

**Comment by team-moeller** [comment]
A fix has been released with version [3.32.04](https://github.com/team-moeller/better-access-charts/releases/tag/3.32.04)

**Reply by Diegomarino80** [reply]
thanks very much for the fix, works fine

I suppose i can't choose schema anymore, it's a shame, my pie colors are almost the same
![image](https://github.com/team-moeller/better-access-charts/assets/146032892/ec143dde-c0ef-4ba7-8f33-61b360835d5a)

**Comment by team-moeller** [comment]
I just released version [3.33.02](https://github.com/team-moeller/better-access-charts/releases/tag/3.33.02).
Please check if this will fix your problem.

**Comment by team-moeller** [comment]
Hi @Diegomarino80 ,
did you had a chance to test the new version.
Please leave an answer here so we can close this discussion

**Reply by Diegomarino80** [reply]
yes, i'm sorry, i was out for work. 
It's a great improvement, thanks very much
